pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -126,7 +132,11 @@ let
     depsBuildBuild = [ buildPackages.stdenv.cc ];
     nativeBuildInputs = [ perl gmp libmpc mpfr ]
       ++ lib.optionals (lib.versionAtLeast version "4.16") [ bison flex ]
-      ++ lib.optional (lib.versionAtLeast version "5.2") pahole;
+      ++ lib.optional (lib.versionAtLeast version "5.2") pahole
+      ++ lib.optionals withRust [ rust-bindgen rustfmt rustc ]
```

Comment:
We don't check the kernel config for any other dependency, so maybe we could just do the version + arch check here as well?

Reply:
I don't like the idea that rust becomes an unconditional dependency of the kernel (yet). When building a kernel with a minimal or custom config, users should not have to download the rust compiler.

Reply:
Also, when I'm building my asahi kernel for aarch64-linux, I want to have the option to enable the rust dependency even though it is not x86_64 (what is supported by upstream for now), since the asahi kernel contains patches to enable rust support for aarch64.